react/components/ModalDialog/index.tsx

```diff
@@ -0,0 +1,100 @@
+import React, { FC, useEffect } from 'react'
+import PropTypes from 'prop-types'
+
+import Modal, { Props as ModalProps } from '../Modal'
+import Button from '../Button'
+
+type DialogOption = {
+  label: string
+  onClick: () => unknown
+}
+
+type Props = Pick<
+  ModalProps,
+  'children' | 'isOpen' | 'onClose' | 'onCloseTransitionFinish' | 'container'
+> & {
+  loading?: boolean
+  confirmation: DialogOption & { isDangerous?: boolean | null }
```

Comment:
Sorry, didn't see this before, why not instead of using a boolean receive just the `variation` from the button? Or if you really want to restrict the options here you could
```suggestion
  confirmation: DialogOption & { variation?: 'primary' | 'danger' }
```

This way we won't have to create something like `isSecondary` later on

Reply:
I agree but I just maintained like it was before because this component will be deprecated soon and I don't pretend to add new things or change it